src/walletSdk/Watcher/index.ts

```diff
@@ -0,0 +1,212 @@
+import isEqual from "lodash/isEqual";
+
+import { Anchor } from "walletSdk/Anchor";
+import { TransactionStatus, WatcherResponse } from "./Types";
+
+interface WatchAllTransactionsRegistry {
+  [assetCode: string]: boolean;
+}
+
+interface TransactionsRegistryAsset {
+  [id: string]: any; // TOOD - replace with Transaction type
+}
+
+interface TransactionsRegistry {
+  [assetCode: string]: TransactionsRegistryAsset;
+}
+
+export class Watcher {
+  private anchor: Anchor;
+
+  private _allTransactionsWatcher?: ReturnType<typeof setTimeout>;
+  private _watchAllTransactionsRegistry: WatchAllTransactionsRegistry;
+  private _transactionsRegistry: TransactionsRegistry;
+  private _transactionsIgnoredRegistry: TransactionsRegistry;
+
+  constructor(anchor: Anchor) {
```

Comment:
Can you add a comment that this class should not be created directly, and instead `.watcher()` on anchor object should be used?

Reply:
Done! I used the same comment pattern from Anchor class declaration